daemon/kmd/api/v1/handlers.go

```diff
@@ -858,6 +858,64 @@ func postTransactionSignHandler(ctx reqContext, w http.ResponseWriter, r *http.R
 	successResponse(w, resp)
 }
 
+// postProgramSignHandler handles `POST /v1/program/sign`
+func postProgramSignHandler(ctx reqContext, w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /v1/program/sign SignProgram
+	//---
+	//    Summary: Sign program
+	//    Description: >
+	//      Signs the passed program with a key from the wallet, determined
+	//      by the account named in the request.
+	//    Produces:
+	//    - application/json
+	//    Parameters:
+	//      - name: Sign Program Request
+	//        in: body
+	//        required: true
+	//        schema:
+	//          "$ref": "#/definitions/SignProgramRequest"
+	//    Responses:
+	//      "200":
+	//        "$ref": "#/responses/SignProgramResponse"
+	var req kmdapi.APIV1POSTProgramSignRequest
+
+	// Decode the request
+	decoder := protocol.NewJSONDecoder(r.Body)
+	err := decoder.Decode(&req)
+	if err != nil {
+		errorResponse(w, http.StatusBadRequest, errCouldNotDecode)
```

Comment:
nit: it would be nice to have "what we failed to decode" in our log files so that we'll have something to work with when we attempt to fix this issue.